### PR TITLE
Extend CodeShovel to capture method annotations

### PR DESCRIPTION
**Pull request by ishtiaque05**
Related to #1
Extend `CodeShovel` to capture methods annotation such as `@Override`, `@Test`, `@SuppressedWarnings` etc.

### CODE REVIEW

**Comment by braxtonhall** [comment]
@ishtiaque05 Can you pull develop into this and add a stub in `PythonFunction` so it builds? I'll fill it out :)

**Comment by braxtonhall** [comment]
Also, do you have a test case for this? Just a single one in the java oracles would be super helpful I think.

File: src/main/java/com/felixgrund/codeshovel/tasks/AnalysisTask.java [hunk 1]
```diff
@@ -330,4 +331,10 @@ public StartEnvironment getStartEnv() {
 	public int getNumCommitsTotal() {
 		return numCommitsTotal;
 	}
+
+	public  void setFunctionAnnotation(String annotations) {
+		this.functionAnnotation = annotations;
+	}
+
+	public String getFunctionAnnotation() { return functionAnnotation; }
```
Comment (braxtonhall) [inline src/main/java/com/felixgrund/codeshovel/tasks/AnalysisTask.java:339]: sorry for the nitpick but can we also move this return statement to the next line if you're gonna be adding commits anyway?

Comment (ishtiaque05) [reply src/main/java/com/felixgrund/codeshovel/tasks/AnalysisTask.java:339]: Oopsie sorry about that 👍🏾 Thanks

**Comment by ishtiaque05** [comment]
> @ishtiaque05 Can you pull develop into this and add a stub in `PythonFunction` so it builds? I'll fill it out :)

Sure no problem :)

**Comment by ishtiaque05** [comment]
> Also, do you have a test case for this? Just a single one in the java oracles would be super helpful I think.

Yes, I will write some as soon as I can execute `mvn test` on my laptop. Apparently, my 8 GB ram is not enough to execute all of the CodeShovel tests. So I would not know if my code breaks anything or not but I can surely write some separate test cases :)

**Comment by braxtonhall** [comment]
> > Also, do you have a test case for this? Just a single one in the java oracles would be super helpful I think.
> 
> Yes, I will write some as soon as I can execute `mvn test` on my laptop. Apparently, my 8 GB ram is not enough to execute all of the CodeShovel tests. So I would not know if my code breaks anything or not but I can surely write some separate test cases :)

Haha well if you want I can test for you. Just give me a test case and I can run it :)

**Pull request merged**